### PR TITLE
feature(mycollection): add artist name filter

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Feature**

This PR resolves []

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- artist name filter in mycollection - pavlos

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (artist name filter in mycollection - pavlos)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against aec38e77c748a4c1f818c0719223bacb55109f8f
</p>

File: src/lib/Components/ArtworkFilter/Filters/ArtistIDsArtworksOptions.tsx [hunk 1]
```diff
@@ -22,21 +22,28 @@ export const ArtistIDsArtworksOptionsScreen: React.FC<ArtistIDsArtworksOptionsSc
...
-      return {
-        displayText: aggCount.name,
-        paramName: FilterParamName.artistIDs,
-        paramValue: aggCount.value,
-        filterKey: "artist",
-      }
-    }) ?? []
+  let artistDisplayOptions
+  if (filterType === "local") {
+    artistDisplayOptions = filterOptions!.find((o) => o.filterType === "artistIDs")?.values!
```
Comment (pvinis) [inline src/lib/Components/ArtworkFilter/Filters/ArtistIDsArtworksOptions.tsx:34]: same as we said in KS. if its `"local"`, we use `values`.

File: src/lib/Components/ArtworkGrids/InfiniteScrollArtworksGrid.tsx [hunk 2]
```diff
@@ -467,8 +467,8 @@ export const InfiniteScrollMyCollectionArtworksGridContainer = createFragmentCon
             image {
               aspectRatio
             }
-            artistNames
             medium
+            artistNames
```
Comment (pvinis) [inline src/lib/Components/ArtworkGrids/InfiniteScrollArtworksGrid.tsx:471]: just moving this to show that this is needed here for the filter.

File: src/lib/Scenes/MyCollection/MyCollection.tsx [hunk 3]
```diff
@@ -337,6 +352,7 @@ export const MyCollectionContainer = createPaginationContainer(
             node {
               id
               medium
+              artistNames
```
Comment (pvinis) [inline src/lib/Scenes/MyCollection/MyCollection.tsx:355]: and also needed here.

File: src/lib/Scenes/MyCollection/MyCollection.tsx [hunk 4]
```diff
@@ -131,6 +131,21 @@ const MyCollection: React.FC<{
...
+          artworks.map((a) => ({
+            displayText: a.artistNames,
+            paramName: FilterParamName.artistIDs,
+            paramValue: a.artistNames,
+          })),
+          (m) => m.paramValue
+        ),
+        // tslint:disable-next-line: no-shadowed-variable
+        localSortAndFilter: (artworks, artists: string[]) => filter(artworks, (a) => artists.includes(a.artistNames)),
+      },
```
Comment (pvinis) [inline src/lib/Scenes/MyCollection/MyCollection.tsx:148]: magic :)

**Pull request closed without merging**